pkg/disk/obd.go

```diff
@@ -0,0 +1,182 @@
+/*
+ * MinIO Cloud Storage, (C) 2020 MinIO, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package disk
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"runtime"
+	"time"
+
+	"github.com/montanaflynn/stats"
+)
+
+const kb = uint64(1 << 10)
+const mb = uint64(kb << 10)
+const gb = uint64(mb << 10)
+
+var globalLatency = map[string]Latency{}
+var globalThroughput = map[string]Throughput{}
+
+// Latency holds latency information for write operations to the drive
+type Latency struct {
+	Avg          float64 `json:"avg_secs,omitempty"`
```

Comment:
For warp I used milliseconds, since it is usually a bit easier to read. Maybe do that?